apps/api/permissions.py

```diff
@@ -67,3 +76,40 @@ class DjangoModelPermissionsWithView(DjangoModelPermissions):
         "PATCH": ["%(app_label)s.change_%(model_name)s"],
         "DELETE": ["%(app_label)s.delete_%(model_name)s"],
     }
+
+
+def verify_hmac(view_func):
+    """Match the HMAC signature in the request to the calculated HMAC using the request payload."""
+
+    # Based on https://github.com/dimagi/commcare-hq/blob/master/corehq/util/hmac_request.py
+    @wraps(view_func)
+    def _inner(request, *args, **kwargs):
+        expected_digest = convert_to_bytestring_if_unicode(request.headers.get("X-Mac-Digest"))
+        secret_key_bytes = convert_to_bytestring_if_unicode(settings.CONNECT_MESSAGING_SERVER_SECRET)
+
+        if not expected_digest and secret_key_bytes:
+            logger.exception(
```

Comment:
I decided to log these as exceptions for now, until the integration is confirmed to be working.